### DISCUSSION TITLE
Failed to configure OBS_VERSION

### DISCUSSION THREAD

**Discussion by andybantly** [obsproject/obs-studio#4535] [Development Questions]
Hello Mods!

I am able to build and run OBS Studio.  My colleague wants to build and run it.  I told him what he needed for prerequisites and those are setup.  I had him clone the repo using Visual Studio.  After cloning, when Visual Studio is parsing, he gets the fatal error "**Failed to configure OBS_VERSION. Either set OBS_VERSION_OVERRIDE or ensure `git describe` succeeds.**"  Does anyone have an idea what may be causing this?

**Comment by notr1ch** [comment]
Read the error, it tells you what to do.

**Reply by andybantly** [reply]
git describe works for me.  I asked him to try that.

**Comment by welocme** [comment]
I have the same problem. If you solve this problem, please tell me the answer

**Reply by stephematician** [reply]
Could be missing the tags in the repository; if you pulled from somewhere other than the origin then (probably) there won't be tags - and then the OBS version isn't correctly identified.

**Comment by Hikari-Tsai** [comment]
That message means the source checkout does not contain enough Git metadata for `git describe` to produce an OBS version. Visual Studio's repository clone can be shallow or omit tags.

From the repository root, verify and repair it with:

```powershell
git fetch --tags --force
git describe --always --tags
git submodule update --init --recursive
```

If it is a shallow clone, unshallow it first:

```powershell
git fetch --unshallow --tags
```

Then delete the CMake cache/build directory and configure again. Cloning according to OBS's build instructions with submodules is preferable to Visual Studio's generic clone workflow.

For a source archive or another environment intentionally lacking `.git`, provide a valid semantic version explicitly:

```powershell
cmake --preset windows-x64 -DOBS_VERSION_OVERRIDE=31.0.0
```

Use the version corresponding to the checked-out source; the override bypasses version discovery but does not replace missing submodules or dependencies.
